src/PIL/Image.py

```diff
@@ -2184,12 +2225,13 @@ def thumbnail(self, size, resample=BICUBIC):
         if size == self.size:
             return
 
-        res = self.draft(None, size)
-        if res is not None:
-            box = res[1]
+        if max_reduce is not None:
+            res = self.draft(None, (size[0] * max_reduce, size[1] * max_reduce))
```

Comment:
On the one hand, `thumbnail()` now takes `max_reduce` into account when working with JPEGs and achieves indistinguishable quality like when it works with any other images.